boards/px4/fmu-v6c/src/i2c.cpp

```diff
@@ -36,5 +36,20 @@
 constexpr px4_i2c_bus_t px4_i2c_buses[I2C_BUS_MAX_BUS_ITEMS] = {
 	initI2CBusExternal(1),
 	initI2CBusExternal(2),
-	initI2CBusInternal(4),
+	initI2CBusExternal(4),
 };
+
+#ifdef BOARD_OVERRIDE_I2C_TREAT_SENSOR_AS_INTERNAL
+bool i2c_treat_sensor_as_internal(uint32_t device_id)
+{
+	// The internal baro and mag on Pixhawk 6C are on an external
+	// bus. On rev 0, the bus is actually exposed externally, on
+	// rev 1+, it is properly internal, however, still marked as
+	// external for compatibility.
+	if (device_id == 396321 || device_id == 4028193) {
```

Comment:
I would use the device type and other constants so this is a little more readable (and potentially less brittle if something changes).

 - DRV_MAG_DEVTYPE_IST8310
 - particular i2c bus
 - particular i2c address
 - etc

Reply:
Yes, I can try to do that. Let's see if I can solve the puzzle.